### DISCUSSION TITLE
Failed to open the URL of my Node.JS app

### DISCUSSION THREAD

**Discussion by remoteconn-7891** [community/community#76917] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

So I'm having trouble accessing the URL of the login page, http://localhost:3000/sign-in, of my Node.JS app. I just keep getting an error. I believe there are no errors with my code in index.JS as far as I know. When I run & debug it, it clearly shows that it's listening on PORT 3000 and it is connected to MongoDB. The only thing I can think of is Mongoose that's causing this issue. PS, I omitted my MongoDB password/username for security purposes. I'll copy/paste the code here and send a few screenshots:

`const bcrypt = require("bcrypt")
const bodyParser = require("body-parser");
const express = require("express");
const passport = require("passport");
const mongoose = require("mongoose");
const LocalStrategy = require("passport-local").Strategy;
const passportLocalMongoose = require("passport-local-mongoose");
const User = require("./model/user");

const app = express();
//ejs extension
var ejs = require('ejs');
app.set('view engine', 'ejs'); 

mongoose.connect("")
.then(() => {
    console.log("MongoDB connection successful:");
  })
  .catch((err => {
    console.error("MongoDB connection error:", err);
  }));

app.use(passport.initialize());
app.use(passport.session());

passport.use(new LocalStrategy ({
    usernameField: 'email',
    passwordField: 'passwd'
},
function(username, password, done) {

}
));
//=======
// Routes
//=======

// Home screen
app.get("/", function (require, res) {
    res.render("home");
});

// Create an account
app.post("/signup", async(req, res) => {
    // Body parse this with middleware functions
    const email = req.body.email;
    const plaintextPassword = req.body.password;

    try {
        // Hash the password with salt
        const saltRounds = 10;
        const hashedPassword = await bcrypt.hash(plaintextPassword, saltRounds);

        // Minimum password length of 8 characters
        let regex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
        let password = "P@ssw0rd";
        let result = regex.test(password);

        if (!isPasswordValid) {
            return res.status(400).json({ error: 'Invalid passowrd format'});
        }

        // Able to create hashed password if valid email
        const user = await User.create({
            email: email,
            password: hashedPassword,
        })
    } catch (error) {
        console.error("Error creating user:", error);
        return res.status(500).json({ error: 'Internal server error' });
    }

    //isEmail function for email validation
    if (isEmail(email)) {
        //If valid email, then can create an account
        const user = await User.create({
            email: email,
            password: req.body.password
        });
    
        //Send reponse after successfully creating user account
        return res.status(200).json(user);
    } else {
        //response for invalid email
        return res.status(400).json({ error: 'Invalid email format' });
    }
    });

//Email validation using Regex
function isEmail(email) {
    var emailFormat = /^[a-zA-Z0-9_.+]+(?<!^[0-9]*)@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
    if (email !== '' && email.match(emailFormat)) { return true; }
    return false;
}

// Sign-In Screen
app.get("/sign-in", function(req, res) {
    res.render("sign-in");
}); 

// User sign-in function
app.post("/sign-in", passport.authenticate("local", {
    successRedirect: "/secret",
    failureRedirect: "/sign-in"
}));

// User Sign-Out function
app.get("/signout", function (req, res) {
    req.logout(); 
        res.redirect('/');
})

    
function isLoggedIn(req, res, next) {
    if (req.isAuthenticated()) return next();
    res.redirect("/sign-in");
}
// Server is listening on PORT 3000
app.listen(process.env.PORT || 3000, function () {
    console.log("Server is listening on PORT 3000");
});`

**Comment by remoteconn-7891** [comment]
<img width="1440" alt="Screenshot 2023-11-24 at 5 53 37 PM" src="https://github.com/community/community/assets/142646120/7c3e727b-eb38-4786-977e-67a0a8387ebd">
<img width="1440" alt="Screenshot 2023-11-24 at 5 53 50 PM" src="https://github.com/community/community/assets/142646120/5e2cc085-9dd3-44b4-87c8-31a9e02ef061">

**Comment by ethansmith345** [comment]
> ### Select Topic Area
> Product Feedback
> 
> ### Body
> So I'm having trouble accessing the URL of the login page, https://smartplayapk.me/, of my Node.JS app. I just keep getting an error. I believe there are no errors with my code in index.JS as far as I know. When I run & debug it, it clearly shows that it's listening on PORT 3000 and it is connected to MongoDB. The only thing I can think of is Mongoose that's causing this issue. PS, I omitted my MongoDB password/username for security purposes. I'll copy/paste the code here and send a few screenshots:
> 
> `const bcrypt = require("bcrypt") const bodyParser = require("body-parser"); const express = require("express"); const passport = require("passport"); const mongoose = require("mongoose"); const LocalStrategy = require("passport-local").Strategy; const passportLocalMongoose = require("passport-local-mongoose"); const User = require("./model/user");
> 
> const app = express(); //ejs extension var ejs = require('ejs'); app.set('view engine', 'ejs');
> 
> mongoose.connect("") .then(() => { console.log("MongoDB connection successful:"); }) .catch((err => { console.error("MongoDB connection error:", err); }));
> 
> app.use(passport.initialize()); app.use(passport.session());
> 
> passport.use(new LocalStrategy ({ usernameField: 'email', passwordField: 'passwd' }, function(username, password, done) {
> 
> } )); //======= // Routes //=======
> 
> // Home screen app.get("/", function (require, res) { res.render("home"); });
> 
> // Create an account app.post("/signup", async(req, res) => { // Body parse this with middleware functions const email = req.body.email; const plaintextPassword = req.body.password;
> 
> ```
> try {
>     // Hash the password with salt
>     const saltRounds = 10;
>     const hashedPassword = await bcrypt.hash(plaintextPassword, saltRounds);
> 
>     // Minimum password length of 8 characters
>     let regex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
>     let password = "P@ssw0rd";
>     let result = regex.test(password);
> 
>     if (!isPasswordValid) {
>         return res.status(400).json({ error: 'Invalid passowrd format'});
>     }
> 
>     // Able to create hashed password if valid email
>     const user = await User.create({
>         email: email,
>         password: hashedPassword,
>     })
> } catch (error) {
>     console.error("Error creating user:", error);
>     return res.status(500).json({ error: 'Internal server error' });
> }
> 
> //isEmail function for email validation
> if (isEmail(email)) {
>     //If valid email, then can create an account
>     const user = await User.create({
>         email: email,
>         password: req.body.password
>     });
> 
>     //Send reponse after successfully creating user account
>     return res.status(200).json(user);
> } else {
>     //response for invalid email
>     return res.status(400).json({ error: 'Invalid email format' });
> }
> });
> ```
> 
> //Email validation using Regex function isEmail(email) { var emailFormat = /^[a-zA-Z0-9_.+]+(?<!^[0-9]*)@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+$/; if (email !== '' && email.match(emailFormat)) { return true; } return false; }
> 
> // Sign-In Screen app.get("/sign-in", function(req, res) { res.render("sign-in"); });
> 
> // User sign-in function app.post("/sign-in", passport.authenticate("local", { successRedirect: "/secret", failureRedirect: "/sign-in" }));
> 
> // User Sign-Out function app.get("/signout", function (req, res) { req.logout(); res.redirect('/'); })
> 
> function isLoggedIn(req, res, next) { if (req.isAuthenticated()) return next(); res.redirect("/sign-in"); } // Server is listening on PORT 3000 app.listen(process.env.PORT || 3000, function () { console.log("Server is listening on PORT 3000"); });`
